AV-101448 - Documentation update all datasources (#328)
couchbasecloud/terraform-provider-couchbase-capella: docs/data-sources/event.md

page_title: "couchbase-capella_event Data Source - terraform-provider-couchbase-capella"
subcategory: ""
description: |-
  Data source for an event in a Capella organization. Events represent a trail of actions that users performs within Capella at an organization level.
---

# couchbase-capella_event (Data Source)

> Data source for an event in a Capella organization. Events represent a trail of actions that users performs within Capella at an organization level.

Reviewer: ```suggestion
The data source to retrieve event details in an organization. Events represent a trail of actions that users performs within Capella at an organization level.
```

## Example Usage

```terraform
data "couchbase-capella_event" "existing_event" {
  organization_id = "<organization_id>"
  id              = "<event_id>"
}
```

<!-- schema generated by tfplugindocs -->
## Schema

### Required

> - `id` (String) UUID for this instance of an Event.

Reviewer: Was this suppose to be eventID? That would change the copy. I'm a bit confused at what ID this is referring to. :)

> - `id` (String) UUID for this instance of an Event.

Author: @Rebecca-Martinez007  No id is correct, when retrieving a resource we return it as "id" in the response. This is different from "eventId" which we use to GET/Update/Delete a particular resource. Check the other GET responses, its always "id" :)
<img width="826" alt="Screenshot 2025-05-16 at 11 40 55 AM" src="https://github.com/user-attachments/assets/cd565531-4975-4274-a297-1efed11f4e5f" />

> - `organization_id` (String) The ID of then Capella organization.

Reviewer: ```suggestion
- `organization_id` (String) The GUID4 ID of the organization.
```

### Read-Only

> - `alert_key` (String) Populated on demand based on the Event.Key and select labels in KV.

Reviewer: ```suggestion
- `alert_key` (String) Generated on demand based on the Event.Key and select labels in KV.
```

> - `app_service_id` (String) SyncGatewayID this Event refers to.

Reviewer: Is this referring to the Sync Gateway instance ID or the App Service ID?

Author: This could be an enhancement, we should discuss with the PM and team to decide on this as it needs to be changed on the management API docs as well. We can take it up separately :)

> - `app_service_name` (String) Name of the sync gateway at the time of event emission.

Reviewer: @PaulomeeCb Is this referring to the Sync Gateway instance name or the App Service name?

Author: This could be an enhancement, we should discuss with the PM and team to decide on this as it needs to be changed on the management API docs as well. We can take it up separately :)

> - `cluster_id` (String) ClusterID this Event refers to.

Reviewer: ```suggestion
- `cluster_id` (String) The GUID4 of the cluster.
```

- `cluster_name` (String) Name of the cluster at the time of event emission.

> - `image_url` (String) Rendered chart image to display for an Alert Event.

Reviewer: ```suggestion
- `image_url` (String) URL to a rendered chart image representing the alert event.
```

> - `incident_ids` (Set of String) Group events related to an alert incident.

Reviewer: @PaulomeeCb is this representing a group of IDs?

- `key` (String) Defines the specific kind of Event.
- `kv` (String) Key-value pairs for additional event data.
- `occurrence_count` (Number) Number of times the alert has fired within this "incident".

> - `project_id` (String) ProjectID this Event refers to.

Reviewer: ```suggestion
- `project_id` (String) The GUID4 ID of the project. 
```

- `project_name` (String) Name of the project at the time of event emission.

> - `request_id` (String) RequestID for an Event.

Reviewer: ```suggestion
- `request_id` (String) The request ID for an event.
```

> - `session_id` (String) User that initiated the request for this Event.

Reviewer: ```suggestion
- `session_id` (String) ID of the user that initiated the request for this event.
```

Reviewer: @PaulomeeCb I'm assuming this is referring to the ID of the user. But it could be "name". I just want to confirm my assumption was right.

Author: @Rebecca-Martinez007 ID is correct, its basically the ID of the user's current session. UserId remains same throughout, the sessionId would change per session. I'll make the description more clear.

- `severity` (String) Severity of the event.
- `source` (String) Identifies the originator of the event.
- `summary` (String) Metadata.SummaryTemplate rendered for this event.

> - `timestamp` (String) Time when the event was emitted.

Reviewer: @PaulomeeCb Is this in RFC3339 format?

- `user_email` (String) Email of the associated user at the time of event emission.

> - `user_id` (String) User id that initiated the request for this Event.

Reviewer: ```suggestion
- `user_id` (String) User ID that initiated the request for this event.
```
